geonode/assets/views.py

```diff
@@ -56,6 +61,11 @@ class AssetViewSet(DynamicModelViewSet):
     serializer_class = AssetSerializer  # TODO: appropriate Serializer should be switched for each Asset instance
     pagination_class = GeoNodeApiPagination
 
+    def get_serializer_class(self):
+        if self.request.method == "POST" and "file" in self.request.FILES:
+            return LocalAssetSerializer
```

Comment:
Pls don't instantiate directly the serializer type, since the system may be configured to use other kinds of Assets.
There is a registry call to get the configured Asset type:
https://github.com/GeoNode/geonode/blob/0b50d16d83df5a409df34a2d49e44419d04b9ecb/geonode/assets/handlers.py#L77
Pls create the new Asset by using the generic AssetHandler methods.